### ISSUE TITLE
N5: sometimes battery drains quickly, device gets hot, top shows ofonod at 100% CPU

### ISSUE THREAD

**Issue by AlanGriffiths** [ubports/ubuntu-touch#984]
Sometimes my phone gets to a state that "eats" battery very quickly when not in use. The phone gets hot.

Shutting down (which, in this state often need two tries before the screen goes fully black) and rebooting seems to fix this.

There are screenshots of battery level and "top" showing ofonod at 100% CPU here:

https://forums.ubports.com/post/15446

This may be related: https://together.jolla.com/question/188336/ofonod-process-takes-up-100-cpu-until-i-reboot-my-device/

**Comment by Thieb67k** [comment]
That recently happened with my LGE Nexus 4 OTA-5.

**Comment by UniversalSuperBox** [comment]
Reports online seem to think that forcing ofono to restart is the only way to fix this. I don't like that idea very much...

**Comment by AlanGriffiths** [comment]
@UniversalSuperBox I agree, that's a workaround at best. I much prefer fixing the code (not that I've even looked at the ofono source).

Do "reports online" have any idea what's happening in ofonod? Or what might nudge it into this state?

Is there more a user (like me) could do to inform the search for an answer?

**Comment by UniversalSuperBox** [comment]
Well, it looks like `ofonod` logs in `/var/log/syslog`, so seeing if we can find any hints there would be a good first step.

However, it looks like a similar issue can affect NetworkManager or the Android Wi-Fi service, making them take up 100% of a core. The only thing these all have in common is that they communicate with the Radio firmware -- another OS that runs on Android devices to handle the device radios.

**Comment by lduboeuf** [comment]
in conclusion, is it something that someone could work on or completely closed source piece of software ?

**Comment by Dionio** [comment]
@UniversalSuperBox Hi, just hit pretty similar issue on FP2 - after updating 16.04 to OTA-6 from OTA-5 the device eats 100% cpu on idle immediately after restart, and the problem is - rebooting the device doesn't help - I had to remove back cover and use hairdryer without heating to be able to diagnose the problem, and keep the device on charge all the time :)
The only option works to the moment - "sudo kill -9 NetworkManager" just after reboot - it helps immediately and now the phone is cold as stone. Until next restart, I guess.

I have dev skills - how can I help with collecting information needed?

Strange thing, btw - "About" still claims it has OTA-5 and "Updates" still offer on update to OTA-6, even after I applied this update 3-4 times. Is it possible, that the update is not marked as complete because of NM behaviour? I'm on Stable channel, if it matters.

**Comment by UniversalSuperBox** [comment]
@Dionio, it sounds like something has gone quite wrong for your device... we're on OTA-8. Maybe you could try running through the UBports Installer again? If Wipe is unchecked (the default) it'll just put Ubuntu Touch on your device again without touching any of your data.

**Comment by Flaburgan** [comment]
Just to be sure, did you activate data sharing? Even if turned off after having it turned on. If so, see https://github.com/ubports/ubuntu-touch/issues/42.
